### PR TITLE
Azure compatibility

### PR DESCRIPTION
**Pull request by pietroalbini**
**This PR is not finished and it's currently untested!**

* [x] Put all interactions with the Travis CI API behind traits
* [x] Remove most references of Travis CI in the code
* [x] Test if the generic code actually works
* [x] Implement support for Azure Pipelines

I'll get more time to work on this on June 28th-29th, if anyone wants to finish this earlier please let me know!

---

This PR implements support for analyzing Azure Pipelines logs, and in general decouples the project from the Travis CI API.

cc @TimNN

### CODE REVIEW

**Comment by pietroalbini** [comment]
@TimNN seems like the current codebase doesn't support GitHub Checks but the deployed code does, is there any change you forgot to push? Or am I missing something?

**Comment by TimNN** [comment]
@pietroalbini: Ugh, that's definitely possible, I'll take a look this evening or tomorrow morning. Sorry if this means that some of the work in this PR needs to be redone.

**Comment by TimNN** [comment]
@pietroalbini: I've just pushed the lastest version to https://github.com/rust-lang/rust-log-analyzer/compare/wip-check-api . There were relatively few travis-related changes, so I hope rebasing won't take too much work.

File: src/ci/azure.rs [hunk 1]
```diff
@@ -54,26 +61,45 @@ struct TimelineLog {
...
 impl Job for TimelineRecord {
-    fn id(&self) -> u64 {
-        unreachable!();
+    fn id(&self) -> String {
+        self.id.clone()
     }
 
     fn html_url(&self) -> String {
-        unreachable!();
+        self.log.as_ref().expect("log url").url.clone()
```
Comment (pietroalbini) [inline src/ci/azure.rs]: It's possible to get an HTML URL that points directly to a job:

```
https://dev.azure.com/rust-lang/rust/_build/results?buildId=<ID>&view=logs&jobId=<JOB-UUID>
```

For example:

https://dev.azure.com/rust-lang/rust/_build/results?buildId=1791&view=logs&jobId=a6152bb1-991e-54db-b884-29fab67829cc

File: src/ci/azure.rs [hunk 2]
```diff
@@ -205,16 +250,29 @@ impl Client {
...
+            return None;
+        }
+        e.check_run
+            .external_id
+            .split('|')
+            .nth(1)
+            .and_then(|id| id.parse().ok())
     }
 
     fn build_id_from_github_status(&self, e: &crate::github::CommitStatusEvent) -> Option<u64> {
```
Comment (pietroalbini) [inline src/ci/azure.rs]: Return `None` here?

File: src/ci/azure.rs (hunk 1 above)
Comment (Mark-Simulacrum) [reply src/ci/azure.rs]: I guess we link to both. I personally almost never find this URL useful, but I guess YMMV. Fixed.

**Pull request merged**